src/pysdmx/model/message.py

```diff
@@ -36,7 +36,7 @@ class ActionType(Enum):
     Information = "information"
 
 
-class Header(Struct, frozen=True, kw_only=True):
```

Comment:
When `frozen` is set to `True`, `msgspec` automatically generates a `__hash__` method, which is something that can be useful from time to time. However, it also disables attribute assignment, which can be inconvenient.

If you think the `__hash__` method can be useful for this class but you still want to be able to set attribute values after the object has been created, you can use the [msgspec.structs.replace function](https://jcristharif.com/msgspec/api.html#msgspec.structs.replace).

So, I'm fine with the removal of the `frozen` flag, but I thought I should mention this, just in case you were not aware of this option and find it better.

Reply:
Yes I do need to edit this class with the dataset references, at this moment it is not necessary to have it frozen

Reply:
To give more context, the xml writer will only use this attribute (and modify it) to set the structure references for this particular message, i.e. get the datasets we want to write and use their IDs and dimension at observation